File: llm_outputs/octocoder/GuruSamples/csharp/replace-conditional-with-polymorphism_refactored.cs
public class Bird {
    //...
    public double getSpeed() {
        switch (type) {
            case EUROPEAN:
                return getBaseSpeed();
            case AFRICAN:
                return getBaseSpeed() - getLoadFactor() * numberOfCoconuts;
            case NORWEGIAN_BLUE:
                return isNailed? 0 : getBaseSpeed(voltage);
            default:
                throw new Exception("Should be unreachable");
        }
    }
}

File: llm_outputs/codellama13b/GuruSamples/python/replace-conditional-with-polymorphism_refactored.py
class Bird:
    #...
    def getSpeed(self):
        if self.type == EUROPEAN:
            return self.getBaseSpeed()
        elif self.type == AFRICAN:
            return self.getBaseSpeed() - self.getLoadFactor() * self.numberOfCoconuts
        elif self.type == NORWEGIAN_BLUE:
            return self.getBaseSpeed(self.voltage) if not self.isNailed else 0
        else:
            raise Exception("Should be unreachable")